core/src/test/java/kafka/clients/consumer/AsyncKafkaConsumerIntegrationTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.clients.consumer;
+
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.GroupProtocol;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.RangeAssignor;
+import org.apache.kafka.common.errors.UnsupportedVersionException;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.test.TestUtils;
+import org.apache.kafka.common.test.api.ClusterConfigProperty;
+import org.apache.kafka.common.test.api.ClusterInstance;
+import org.apache.kafka.common.test.api.ClusterTest;
+import org.apache.kafka.common.test.api.ClusterTestExtensions;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+@ExtendWith(ClusterTestExtensions.class)
+public class AsyncKafkaConsumerIntegrationTest {
```

Comment:
We actually have `PlaintextConsumerTest` for consumer's integration tests. Any reason why you did not put those there?

Reply:
The cases in `PlaintextConsumerTest` class use same broker config. However, in `AsyncKafkaConsumerIntegrationTest`, we would like to test two combination:

1. default `group.coordinator.new.enable` with `group.coordinator.rebalance.protocols=classic`
2. `group.coordinator.new.enable=false` with default `group.coordinator.rebalance.protocols`

To accomplish it in `PlaintextConsumerTest`, we need to tear down brokers first and restart brokers with new config. I think it's not convenient in old test framework. Probably, we can create another Jira to migrate `AbstractConsuemrTest` to new framework in the future, so we can support more different broker config cases. What do you think? Thank you.